types/src/transaction/transaction.rs

```diff
@@ -264,6 +232,115 @@ fn is_input_and_output_consistent(inputs: &[AssetTransferInput], outputs: &[Asse
     sum.iter().all(|(_, sum)| sum.is_zero())
 }
 
+fn make_tag(mut item: Vec<u8>, mut result: u128) -> u128 {
+    if item.is_empty() {
+        return result
+    }
+    // Assume that the mask is stored as little endian
+    let head = item.pop().unwrap();
+    result = (result << 8) | head as u128;
+    make_tag(item, result)
+}
+
+
+fn apply_bitmask_to_output(
+    mut bitmask: u128,
+    outputs: Vec<AssetTransferOutput>,
+    mut result: Vec<AssetTransferOutput>,
+) -> Result<Vec<AssetTransferOutput>, HashingError> {
+    for i in 0.. {
+        if bitmask == 0 {
+            return Ok(result)
+        }
+
+        if outputs.is_empty() {
+            return Err(HashingError::InvalidFilter)
+        }
+
+        if (bitmask & 0x1) == 1 {
+            result.push(outputs[i].clone());
+        }
+        bitmask = bitmask >> 1
+    }
+    Err(HashingError::InvalidFilter)
+}
+
+fn apply_sign_scheme_to_input(
+    inputs: &Vec<AssetTransferInput>,
+    scheme: bool,
```

Comment:
Rename scheme to is_sign_all

Reply:
I changed it

Reply:
It's not changed yet.

Reply:
I misunderstand your request. I changed it.